scripts/build.py

```diff
@@ -14,22 +14,38 @@
 
 """Build file for production version of Oppia. Minifies JS and CSS."""
 
+import hashlib
+import json
 import os
 import re
 import shutil
 import subprocess
 import sys
-import yaml
 
-
-HEAD_DIR = os.path.join('core', 'templates', 'dev', 'head', '')
-OUT_DIR = os.path.join('core', 'templates', 'prod', 'head', '')
 REMOVE_WS = re.compile(r'\s{2,}').sub
 YUICOMPRESSOR_DIR = os.path.join(
     '..', 'oppia_tools', 'yuicompressor-2.4.8', 'yuicompressor-2.4.8.jar')
 
-FILE_EXTENSIONS_TO_IGNORE = ['.py']
-
+# Files with this extension shouldn't be moved to build folder.
+FILE_EXTENSIONS_TO_IGNORE = ('.py')
+# Files with this extension should be moved to build folder but shouldn't be
+# renamed (i.e. the filepath shouldn't contain hash)
+# This is because these files are referenced from places we can't influence.
```

Comment:
Can you elaborate briefly here? This sounds a bit cryptic.

Also, where exactly do these files come from? I don't remember having .eot, .woff2, etc. files in our codebase, so I'm a bit confused.